src/Vegetation/PlantHydraulics.jl

```diff
@@ -435,29 +566,36 @@ function flux_out_roots(
     Y::ClimaCore.Fields.FieldVector,
     p::ClimaCore.Fields.FieldVector,
     t::FT,
-)::FT where {FT}
+)::ClimaCore.Fields.Field where {FT}
     @unpack vg_α, vg_n, vg_m, ν, S_s, K_sat, root_distribution =
         model.parameters
-    S_l_stem = effective_saturation(ν, Y.vegetation.ϑ_l[1])
-    ψ_stem = water_retention_curve.(vg_α, vg_n, vg_m, S_l_stem, ν, S_s)
-    sum_flux_out_roots = sum(
-        flux.(
-            model.domain.root_depths,
-            model.domain.compartment_midpoints[1],
-            re.ψ_soil(t),
-            ψ_stem,
-            vg_α,
-            vg_n,
-            vg_m,
-            ν,
-            S_s,
-            K_sat[:root],
-            K_sat[:stem],
-        ) .* root_distribution.(model.domain.root_depths) .* (
-            vcat(model.domain.root_depths, [0.0])[2:end] -
-            vcat(model.domain.root_depths, [0.0])[1:(end - 1)]
-        ),
-    ) # m/s
+    ψ_stem = p.vegetation.ψ[1]
+    n_root_layers = length(model.root_depths)
+    sum_flux_out_roots = similar(ψ_stem)
```

Comment:
We are allocating here, which isnt good, but I think we will come back and address with when we add in the root compartments as prognostic variables.